src/lab1/simulator.java

```diff
@@ -4,107 +4,145 @@
+        packet_size = scanner.nextInt();
+
+        System.out.println("C, Transmission rate (bits/sec): ");
+        final int transmission_rate = scanner.nextInt();
+
+        // (ticks) = ((bits) / (bits / sec)) / (sec / tick)
+        t_transmission = (int) Math.ceil(
+            (packet_size / transmission_rate) / tick_duration
+        );
+
+        q_utilizaton = packet_size * (packet_gen_rate / transmission_rate);
+
+        pick_a_queue();
+    }
+
     public static void arrival() {
-        t = System.currentTimeMillis() - t_start;
         if (t >= t_arrival) {
-            KendallPacket packet = new KendallPacket(L);
-            queue.add(packet);
+            KendallPacket new_packet = new KendallPacket(packet_size);
+
+            queue.add(new_packet);
+
             t_arrival = t + calc_arrival_time();
-            t_departure = t + (L/C)*ms_per_second;// bits/(bits/second)*1000(ms/second) = ms
+            t_departure = t + t_transmission;
             // TODO add support for the packet loss case of MD1K where you can't add to the queue. Maybe do something like have queue.add() return a boolean representing success. If it's false then increase number of packets lost by 1. This would only be false in the situation that you have a MD1K queue and that queue is full and you try to add a packet to it
         }
     }
 
     public static void departure() {
-        t = System.currentTimeMillis() - t_start;
         if (t >= t_departure) {
             queue.pop();
         }
     }
 
     public static long calc_arrival_time() {
         double u = Math.random(); // random number between 0 and 1
-        double arrival_time_in_seconds = ((-1 / lambda) * Math.log(1 - u)); // exponential random variable
-        double arrival_time_in_milliseconds = arrival_time_in_seconds * ms_per_second;
-        return (long)arrival_time_in_milliseconds;
+        double arrival_time =
+            ((-1 / packet_gen_rate) * Math.log(1 - u)) / tick_duration;
+
+        return (long) Math.ceil(arrival_time);
     }
 
     public static void pick_a_queue() {
-        System.out.println("Would you like to simulate using a M/D/1 queue (y) or a M/D/1/K queue (n)? (y/n)");
-        String m_d_one_choice = scanner.next();
-        while (!(m_d_one_choice.equals("y") || m_d_one_choice.equals("n"))) {
-            System.out.println("Please enter 'y' for M/D/1 or 'n' for M/D/1/K: ");
-            m_d_one_choice = scanner.next();
+        final String MD1 = "y";
+        final String MD1K = "n";
+        final String msg = "M/D/1 queue (y) or a M/D/1/K queue (n)? (y/n)";
+
+        System.out.println(msg);
+        String choice = scanner.next();
+
+        while (!(choice.equals(MD1) || choice.equals(MD1K))) {
+            System.out.println(msg);
+            choice = scanner.next();
         }
 
-        if (m_d_one_choice.equals("y")) {
-            is_m_d_one = true;
-            scanner.close();
```

Comment:
add back the scanner.close() line  to the if(choice.equals(MD1)) part. Otherwise, memory leaks